### PR TITLE
Fix compatibility with Docker 25+

### LINKED ISSUE #498: Unable to analyze the image using dive
**What happened**:  I am using RHEL9  free tier from AWS and installed the dive with below steps by referring the github page https://github.com/wagoodman/dive  and the version is dive 0.11.0
i have downloaded the Docker image "nginx:latest" by using "docker pull nginx" command
then i have executed "dive nginx:latest", getting below error

[ec2-user@ip-172-31-36-53 ~]$ sudo dive nginx
Image Source: docker://nginx
Fetching image... (this can take a while for large images)
cannot fetch image
could not find image config
[ec2-user@ip-172-31-36-53 ~]$

Could you please let me know how to scan an image using dive?

**Anything else we need to know?**: I am interested to know how to include the dive into jenkins CI/CD pipeline.  like we are using Free style jobs so need to include the configuration in the "execute shell"

**Environment**: RHEL9
- OS version: 
              - [ec2-user@ip-172-31-36-53 ~]$ uname -a
              Linux ip-172-31-36-53.ec2.internal 5.14.0-362.8.1.el9_3.x86_64 #1 SMP PREEMPT_DYNAMIC Tue Oct 3 11:12:36 EDT 2023 x86_64 x86_64 x86_64 GNU/Linux
              [ec2-user@ip-172-31-36-53 ~]$

- Docker version (if applicable)
              - [ec2-user@ip-172-31-36-53 ~]$ docker --version
              Docker version 25.0.0, build e758fe5
[ec2-user@ip-172-31-36-53 ~]$

### PR DESCRIPTION
**Pull request by tbroyer**
Add support for OCI-compatible Docker images.

Fixes #498

### CODE REVIEW

**Comment by char-iot** [comment]
also fixes #452

**Comment by Kerwood** [comment]
Is it backwards compatible ?

**Comment by mikesir87** [comment]
@Kerwood - I've confirmed this works with both the new format and is backwards compatible

**Comment by mikesir87** [comment]
For those currently stuck (and don't want to build it themselves), I did build and publish it under my namespace (and it's multi-arch!). Image here - `michaelirwin244/dive`

**Review by wagoodman — APPROVED** [review]

**Comment by wagoodman** [comment]
I'm shooting for today

File: dive/image/docker/image_archive.go [hunk 1]
```diff
@@ -82,6 +85,55 @@ func NewImageArchive(tarFile io.ReadCloser) (*ImageArchive, error) {
...
+			} else if strings.HasPrefix(name, "blobs/") {
+				// For the OCI-compatible image format (used since Docker 25), use mime sniffing
+				// but limit this to only the blobs/ (containing the config, and the layers)
+
+				// The idea here is that we try various formats in turn, and those tries should
+				// never consume more bytes than this buffer contains so we can start again.
+
+				// 512 bytes ought to be enough (as that's the size of a TAR entry header),
+				// but play it safe with 1024 bytes. This should also include very small layers
+				// (unless they've also been gzipped, but Docker does not appear to do it)
```
Comment (zevisert) [inline dive/image/docker/image_archive.go:97]: I have a counter-example that's breaking this assumption, here's the layer from an image manifest that I've provided as a repro in #507 

```jsonc
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "config" { ... },
  "layers": [
    { ... },
    { ... },
    {
      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",  // <- This layer is gzipped
      "digest": "sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1",
      "size": 32 // <- this layer is also very small
    }
  ]
}
```

**Pull request merged**